### PR TITLE
Changed the Color struct.

### LINKED ISSUE #45: color has no default alpha value
https://github.com/schveiguy/raylib-d/blob/a8f9ab2cccec720d7a09aff1d42607a9ed305e5e/source/raylib/package.d#L182

setting it to 255 makes it better; hottake

### PR DESCRIPTION
**Pull request by Kapendev**
Resolves #45 #42  
I also added constructors, which required modifying a small part of raygui.

### CODE REVIEW

**Comment by schveiguy** [comment]
Ah, thanks for the PR. However, raylib/package.d is auto-generated and I don't want to put custom code there. This is why e.g. `Vector2` is located in `raylib_types`.

Can you move it there, and then change the `generating.md` appropriately?

File: source/raylib/package.d [hunk 1]
```diff
@@ -179,7 +179,37 @@ struct Color
...
+    this(ubyte r, ubyte g, ubyte b, ubyte a) {
+        this.r = r;
+        this.g = g;
+        this.b = b;
+        this.a = a;
+    }
+
+    this(ubyte r, ubyte g, ubyte b) {
+        this(r, g, b, 255);
+    }
```
Comment (schveiguy) [inline source/raylib/package.d:195]: No need for an overload here.

```suggestion
    this(ubyte r, ubyte g, ubyte b, ubyte a = 255) {
        this.r = r;
        this.g = g;
        this.b = b;
        this.a = a;
    }
```

File: source/raylib/package.d [hunk 2]
```diff
@@ -179,7 +179,37 @@ struct Color
...
+
+    this(uint rgba) {
+        this(
+            (rgba & 0xFF000000) >> 24,
+            (rgba & 0xFF0000) >> 16,
+            (rgba & 0xFF00) >> 8,
+            (rgba & 0xFF),
+        );
+    }
```
Comment (schveiguy) [inline source/raylib/package.d:212]: I get this, but I think this is presuming a format for all `uint` that are passed. This is going cause some odd cases, people might assume something different, or they might accidentally have a typo and pass e.g. `255` in here, and it compiles and does something very odd.

In addition, raylib already has the [`GetColor`](https://github.com/raysan5/raylib/blob/d9c5066382615644137b4f65479c65c44820027a/src/rtextures.c#L4805) function, which does something similar.

I'd rather see factory functions that do this if you have a specific format you want to parse (not in this PR).

**Review by schveiguy — CHANGES_REQUESTED** [review]
Pending the changes, and the move to raylib_types

File: source/raylib/package.d (hunk 2 above)
Comment (Kapendev) [reply source/raylib/package.d]: Yeah, it can create some issues.

**Comment by Kapendev** [comment]
I moved it to `raylib_types` and changed the markdown file too.

**Review by schveiguy — APPROVED** [review]
Looks good! I'm considering whether this will be released as a point release or wait for a major release, because technically it breaks code (the raygui code being an example).

**Comment by schveiguy** [comment]
It was mentioned that maybe Color.init should be all 0s (including alpha)_. I'm thinking we should make that the case, and just have the default parameter for alpha be 255 in the constructor. What do you think?

Making one byte not be zero can have an effect on default initializers in things that contain a Color. When you have to set something to all 0 it's pretty efficient.

**Comment by Kapendev** [comment]
> Looks good! I'm considering whether this will be released as a point release or wait for a major release, because technically it breaks code (the raygui code being an example).

Maybe a point release? I also want to work on the Linear mixin, so maybe these two could go together.

**Comment by Kapendev** [comment]
> It was mentioned that maybe Color.init should be all 0s (including alpha). I'm thinking we should make that the case, and just have the default parameter for alpha be 255 in the constructor. What do you think?
> 
> Making one byte not be zero can have an effect on default initializers in things that contain a Color. When you have to set something to all 0 it's pretty efficient.

a=255 will also break some code. Everything being zero makes more sense if we don't want that. There used to be #42 , but with how things are built now (constructors), that's not an issue anymore. I prefer a=0, but it's a better idea to ask on Discord for that.

**Pull request merged**